app.js

```diff
@@ -0,0 +1,153 @@
+
+    db.collection('users').remove((error, result) => { });
+    db.collection('accounts').remove((error, result) => { });
+    db.collection('users').insert(users, (error, result) => { });
+    db.collection('accounts').insert(accounts, (error, result) => { });
+  });
+  res.send('done');
+});
+
+app.get('/account/:userID/balance', (req, res) => {
+  MongoClient.connect(MONGO_URI, (err, db) => {
+    if(err) {
+      // TODO handle db error
+      res.send('Error connecting to db');
+    }
+    // TODO input validation
+    const userId = parseInt(req.params.userID);
+    
+    getAccountBalance(db, 'accounts', userId)
+      .then(result => {
+        res.send(result);
+      })
+      .catch(error => {
+        res.send('Error getAccountBalance ' + error);
+      });
+    });
+});
+
+app.put('/account/:userID/deposit', (req, res) => {
+  MongoClient.connect(MONGO_URI, (err, db) => {
+    if(err) {
+      // TODO handle db error
+      res.send('Error connecting to db');
+    }
+    // TODO input validation
+    const userId = parseInt(req.params.userID);
+    const depositObject = req.body;
+    const accountType = depositObject.type;
+    const depositAmount = depositObject.amount;
+
+    modifyFunds(db, 'accounts', userId, accountType, depositAmount, 1)
+      .then(result => {
+        getAccountBalance(db, 'accounts', userId)
+        .then(result => {
+          res.send(result);
+        })
+        .catch(error => {
+          res.send('Error getting new account balance ' + error);
+        });
+      })
+      .catch(error => {
+        res.send('Error depositing funds ' + error);
+      });
+  });
+});
+
+app.put('/account/:userID/withdraw', (req, res) => {
+  MongoClient.connect(MONGO_URI, (err, db) => {
+    if(err) {
+      // TODO handle db error
+      res.send('Error connecting to db');
+    }
+    // TODO input validation
+    const userId = parseInt(req.params.userID);
+    const depositObject = req.body;
+    const accountType = depositObject.type;
+    const depositAmount = depositObject.amount;
+
+    modifyFunds(db, 'accounts', userId, accountType, depositAmount, -1)
+      .then(result => {
+        getAccountBalance(db, 'accounts', userId)
+        .then(result => {
+          res.send(result);
+        })
+        .catch(error => {
+          res.send('Error getting new account balance ' + error);
+        });
+      })
+      .catch(error => {
+        res.send('Error withdrawing funds ' + error);
+      });
+  });
+});
+
+function modifyFunds(db, collection, uID, accountType, depositAmount, modifier) {
+  return new Promise((resolve, reject) => {
+
+    const search = {userId: uID, type: accountType};
+    const modify = {$inc: {amount: depositAmount * modifier}}
+    db.collection(collection).update(search, modify, (err, res) => {
+      if(err) {
+        reject(err);
+      }
+      resolve(res);
+    });
+  });
+};
+
+function getAccountBalance(db, collection, uID) {
```

Comment:
Just me, but `uID` is not camelcase when everything else is. I think a lot of people go with just `uid`, but for this block of code I would call it `userId` because then on the line where you do `const search = {userId: uID};` you could do `const search = {userId};`